fedimint-client/src/db.rs

```diff
@@ -262,3 +276,173 @@ impl_db_lookup!(
     key = ClientModuleRecovery,
     query_prefix = ClientModuleRecoveryPrefix
 );
+
+/// ClientMigrationMap is a BTreeMap that maps DatabaseVersions to async
+/// functions. These functions are expected to "migrate" the database from the
+/// keyed DatabaseVersion to DatabaseVersion + 1.
+pub type ClientMigrationMap = BTreeMap<
```

Comment:
I don't like the name, but need more coffee to rationalize why. :D

Would consider wrapping it in a newtype.

Reply:
`BTreeMap<DatabaseVersion, ClientMigrationFn>` is probably more legible.